dulwich/tests/compat/test_repository.py

```diff
@@ -119,3 +127,32 @@ def test_packed_objects(self):
     def test_all_objects(self):
         expected_shas = self._get_all_shas()
         self.assertShasMatch(expected_shas, iter(self._repo.object_store))
+
+
+@skipIf(git_version() < (2, 5), 'Git version must be >= 2.5')
+class WorkingTreeTestCase(ObjectStoreTestCase):
+    """Test for compatibility with git-worktree."""
+
+    def setUp(self):
+        super(WorkingTreeTestCase, self).setUp()
+        self._worktree_path = self.create_new_worktree(self.repo_path())
+        self._worktree_repo = Repo(self._worktree_path)
+        self._mainworktree_repo = self._repo
+        self._repo = self._worktree_repo
+
+    def tearDown(self):
+        self._worktree_repo.close()
+        rmtree_ro(self._worktree_path)
+        self._repo = self._mainworktree_repo
+        super(WorkingTreeTestCase, self).tearDown()
+
+    def test_refs(self):
+        super(WorkingTreeTestCase, self).test_refs()
+        self.assertTrue(self._mainworktree_repo.refs.allkeys()==\
```

Comment:
Fixed in 7b05f09d856a79ada383b6e2b2c514b996cb55d8.

Sorry, I am rather new in Python programming ([I am rather a C++ guru][CGAL]). Thanks for the advice.

[CGAL]: http://github.com/CGAL/cgal